src/aliceVision/camera/IntrinsicBase.hpp

```diff
@@ -201,7 +201,23 @@ inline double AngleBetweenRay(
   const Vec3 ray2 = (pose2.rotation().transpose() * intrinsic2->operator()(x2)).normalized();
   const double mag = ray1.norm() * ray2.norm();
   const double dotAngle = ray1.dot(ray2);
-  return R2D(acos(clamp(dotAngle/mag, -1.0 + 1.e-8, 1.0 - 1.e-8)));
+  return radianToDegree(acos(clamp(dotAngle/mag, -1.0 + 1.e-8, 1.0 - 1.e-8)));
+}
+
+/// Return the angle (degree) between two poses and a 3D point.
+inline double AngleBetweenRay(
+  const geometry::Pose3 & pose1,
+  const geometry::Pose3 & pose2,
+  const Vec3 & pt3D)
+{
+  // Law of cosines
+  const Vec3 center1 = pose1.center();
+  const Vec3 center2 = pose2.center();
+  const double baseline = (center1 - center2).norm();
+  const double ray1 = (pt3D - center1).norm();
```

Comment:
u can just use the dot product of the normal vectors, ie
```
const Vec3 ray1 = (pt3D - center1). normalized();
const Vec3 ray2 = (pt3D - center2). normalized();
return radianToDegree(std::acos(ray1.dot(ray2)));
```
and add the doxygen doc :-)

Reply:
Actually i didn't see the same function above. We can factorize then:
in `numeric.hpp` u can declare
```c++
/**
 * @brief Compute the angle between two 3d vectors.
 * @param[in] ray1 The first vector.
 * @param[in] ray2 The second vector.
 * @return the angle in degrees between the two vectors.
 */
inline double AngleBetweenRays(const Vec3 & ray1, const Vec3 & ray2)
{
  const double bound = 1.0 - 1.e-8;
  const auto dotprod = ray1.normalized().dot(ray2.normalized());
  return radianToDegree(std::acos(std::clamp(dotprod, -bound, bound)));
}
```
and call that inside the two functions (which btw should be named `AngleBetweenRays` (missing s at the end))

Reply:
I didn't see your comment but I did (almost) the same thing :)